src/Razor/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentGenericTypePass.cs

```diff
@@ -118,36 +136,110 @@ private void Process(ComponentIntermediateNode node)
                 //
                 // For example, consider a repeater where the generic type is the 'item' type, but the developer has
                 // not set the items. We won't be able to do type inference on this and so it will just be nonsense.
-                var attributes = node.Attributes.Select(a => a.BoundAttribute).Concat(node.ChildContents.Select(c => c.BoundAttribute));
-                foreach (var attribute in attributes)
+                foreach (var attribute in node.Attributes)
                 {
-                    if (attribute == null)
+                    if (TryFindGenericTypeNames(attribute.BoundAttribute, out var typeParameters))
```

Comment:
Good question. I don't really know why there was a null check in the original code (i.e., if there actually is any case where it could be null). None of the integration tests fail here, so it seems very likely that the data is constructed in a way that won't include null entries inside `node.Attributes`.

However since I don't absolutely know this is always true, I'll put the null check back in. Thanks!